raiden/network/proxies/user_deposit.py

```diff
@@ -360,6 +386,61 @@ def approve_and_deposit(
 
             self._deposit_check_result(transaction_sent, token, total_deposit, amount_to_deposit)
 
+    def plan_withdraw(
+        self, amount: TokenAmount, given_block_identifier: BlockIdentifier
+    ) -> BlockNumber:
+        """ Announce that you plan to withdraw tokens from the UserDeposit contract
+
+        Returns the block number at which the withdraw is ready.
+        """
+        self._plan_withdraw_preconditions(amount, given_block_identifier)
+
+        # Simplify our lives by disallowing concurrent plan_withdraw / withdraw calls
+        with self._withdraw_lock:
+            estimated_transaction = self.client.estimate_gas(
+                self.proxy, "planWithdraw", {}, amount
+            )
+            transaction_sent = None
+            if estimated_transaction is not None:
+                transaction_sent = self.client.transact(estimated_transaction)
+            transaction_mined = self._plan_withdraw_check_result(
+                transaction_sent=transaction_sent, amount_to_plan_withdraw=amount
+            )
+
+        assert transaction_mined is not None, "_plan_withdraw_check_result returned None"
+
+        return BlockNumber(transaction_mined.receipt["blockNumber"] + self.get_withdraw_delay())
+
+    def withdraw(self, amount: TokenAmount, given_block_identifier: BlockIdentifier) -> None:
+        """ Withdraw tokens from UDC, requires a mature withdraw plan"""
+
+        token_address = self.token_address(given_block_identifier)
+        token = self.proxy_manager.token(
```

Comment:
note that we have a runtime cycle here, one of these references should be weak. (I knot this was not introduced by this PR, I'm just pointing  it out)